azurerm/resource_arm_app_service_virtual_network_connection.go

```diff
@@ -0,0 +1,513 @@
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"kind": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"properties": {
+										Type:     schema.TypeList,
+										MaxItems: 1,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"start_address": {
+													Type:     schema.TypeString,
+													Optional: true,
+												},
+
+												"end_address": {
+													Type:     schema.TypeString,
+													Optional: true,
+												},
+
+												"route_type": {
+													Type:     schema.TypeString,
+													Optional: true,
+													Default:  "DEFAULT",
+													ValidateFunc: validation.StringInSlice([]string{
+														string(web.DEFAULT),
+														string(web.INHERITED),
+														string(web.STATIC),
+													}, true),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+
+						"resync_required": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"dns_servers": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceVirtualNetworkConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Web.AppServicesClient
+	vnetClient := meta.(*ArmClient).Network.VnetClient
+	vnetGatewayClient := meta.(*ArmClient).Network.VnetGatewayClient
+
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM App Service Virtual Network Connection creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	appServiceName := d.Get("app_service_name").(string)
+
+	properties := d.Get("properties").([]interface{})
+	property := properties[0].(map[string]interface{})
+	vnetResourceId := property["vnet_resource_id"].(string)
+
+	id, err := azure.ParseAzureResourceID(vnetResourceId)
+	if err != nil {
+		return err
+	}
+	vnetResGroup := id.ResourceGroup
+	vnetName := id.Path["virtualNetworks"]
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.GetVnetConnection(ctx, resGroup, appServiceName, vnetName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("error checking for presence of existing App Service Virtual Network Connection for app %q vnet %q (Resource Group %q): %s", appServiceName, vnetName, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_app_service_virtual_network_connection", *existing.ID)
+		}
+	}
+
+	if err := checkAppServiceExist(client, ctx, &resGroup, &appServiceName); err != nil {
```

Comment:
Why are we checking for the existence of these resources instead of letting the API do it?